code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/caching/CacheFileManager.java

```diff
@@ -0,0 +1,290 @@
+                                            final String metadataFilePath) {
+        if (cacheEntry == null || StringUtils.isNullOrEmpty(metadataFilePath)) {
+            return false;
+        }
+
+        final Map<String, String> metadata = new HashMap<>();
+        metadata.put(FileCacheResult.METADATA_KEY_PATH_TO_FILE, entryLocation);
+
+        final Date expiry = cacheEntry.getExpiry().getExpiration();
+        if (expiry != null) {
+            metadata.put(FileCacheResult.METADATA_KEY_EXPIRY_IN_MILLIS, String.valueOf(expiry.getTime()));
+        }
+
+        if (cacheEntry.getMetadata() != null) {
+            metadata.putAll(cacheEntry.getMetadata());
+        }
+
+        try {
+            final JSONObject metadataJson = new JSONObject(metadata);
+            final InputStream metadataStream =
+                    new ByteArrayInputStream(metadataJson.toString().getBytes(StandardCharsets.UTF_8));
+            return FileUtils.readInputStreamIntoFile(new File(metadataFilePath),
+                    metadataStream, false);
+        } catch (final Exception exception) {
+            Log.debug(ServiceConstants.LOG_TAG, TAG, "Cannot create cache metadata %s", exception);
+            return false;
+        }
+    }
+
+    /**
+     * Deletes the cache file and a cache metadata file associated with {@code key}.
+     *
+     * @param cacheName the cache sub-folder from which files associated with the key need to be deleted
+     * @param key       the key for which the cache file and cache metadata file should be deleted
+     * @return true if the cache file and metadata file were deleted successfully; false otherwise.
+     */
+    boolean deleteCacheFile(final String cacheName, final String key) {
+        if (StringUtils.isNullOrEmpty(cacheName) || StringUtils.isNullOrEmpty(key)) {
+            return false;
+        }
+
+        final File cacheFileToDelete = getCacheFile(cacheName, key);
+
+        if (cacheFileToDelete == null) {
+            Log.debug(ServiceConstants.LOG_TAG, TAG, "Cannot delete cache file. No file to delete.");
+            return true;
+        }
+
+        if (!FileUtils.deleteFile(cacheFileToDelete, true)) {
+            Log.debug(ServiceConstants.LOG_TAG, TAG, "Failed to delete cache file for " +
+                    "cache name [%s], key: [%s]", cacheName, key);
+            return false;
+        }
+
+        final String cacheFileMetadataPath = getMetadataLocation(cacheName, key);
+        if (cacheFileMetadataPath != null) {
+            Log.debug(ServiceConstants.LOG_TAG, TAG, "Failed to delete cache metadata file for " +
+                    "cache name [%s], key: [%s]", cacheName, key);
+            FileUtils.deleteFile(new File(cacheFileMetadataPath), true);
+        }
+
+        return true;
+    }
+
+    @VisibleForTesting
+    String getCacheLocation(final String cacheName, final String key) {
+        if (StringUtils.isNullOrEmpty(cacheName) || StringUtils.isNullOrEmpty(key)) {
```

Comment:
nit: I notice these checks are redundant as the public API does similar checks. Can you mark the parameters `NonNull` and remove these redundant checks?